src/component/handlers/edit/editOnInput.js

```diff
@@ -42,22 +43,56 @@ function editOnInput(editor: DraftEditor): void {
 
   // We have already updated our internal state appropriately for this input
   // event. See editOnBeforeInput() for more info
-  if (editor._waitingOnInput) {
-    if (!editor._renderNativeContent) {
-      return;
+  if (editor._updatedNativeInsertionBlock && !editor._renderNativeContent) {
+    editor._updatedNativeInsertionBlock = false;
+    return;
+  }
+
+  editOnSelect(editor);
+  var editorState = editor._latestEditorState;
+
+  if (editor._updatedNativeInsertionBlock) {
+    const oldBlock = editor._updatedNativeInsertionBlock;
+    if (editorState.getSelection().getAnchorKey() !== oldBlock.getKey()) {
```

Comment:
How does this play with the case in which the mouse is down when we enter `onBeforeInput`? Is it possible to drag the mouse down between `onBeforeInput` and here so that the selection changes from being collapsed to not collapsed? In particular, can the focus change while the anchor does not?